### PR TITLE
new file for venture drill dash cooldown

### PR DESCRIPTION
**Pull request by vwade**
proposed fix for venture's 6 second cooldown

    - live build cd: 8
    - tested cd: 6

    this is a small patch which will constrain venture's cooldown to 6 seconds

    - new file added: venture/drill_dash.opy

    * tested with developer build (practice range)
    * tested on internal developer build

### CODE REVIEW

File: src/heroes/venture/drill_dash.opy [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!mainFile "../../dev_main.opy"
+
+#!define ADJ_VENTURE_DRILL_DASH_COOLDOWN_BURROWED ADJ_VENTURE_DRILL_DASH_COOLDOWN / 2
```
Comment (MaxwellJung) [inline src/heroes/venture/drill_dash.opy:3]:
```suggestion
#!define ADJ_VENTURE_DRILL_DASH_COOLDOWN_BURROWED (ADJ_VENTURE_DRILL_DASH_COOLDOWN/2)
```

Comment (MaxwellJung) [inline src/heroes/venture/drill_dash.opy]:
```suggestion
#!define ADJ_VENTURE_DRILL_DASH_COOLDOWN_BURROWED (ADJ_VENTURE_DRILL_DASH_COOLDOWN/2)
```

in-line math should be surrounded with parenthesis

File: src/heroes/venture/drill_dash.opy [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+rule "[venture/drill_dash.opy]: Constrain Drill Dash Cooldown":
+    @Event eachPlayer
+	@Condition eventPlayer.getCurrentHero() == Hero.VENTURE
+    @Condition not 0 == eventPlayer.getAbilityCooldown(Button.SECONDARY_FIRE)
+	
+	if eventPlayer.isUsingAbility1():
+		eventPlayer.setAbilityCooldown(Button.SECONDARY_FIRE, ADJ_VENTURE_DRILL_DASH_COOLDOWN_BURROWED)
+	else:
+		eventPlayer.setAbilityCooldown(Button.SECONDARY_FIRE, ADJ_VENTURE_DRILL_DASH_COOLDOWN)
```
Comment (MaxwellJung) [inline src/heroes/venture/drill_dash.opy]:
```suggestion
		eventPlayer.setAbilityCooldown(Button.SECONDARY_FIRE, ADJ_VENTURE_DRILL_DASH_COOLDOWN)

```

add new line at EOF

File: src/heroes/venture/drill_dash.opy [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!mainFile "../../dev_main.opy"
+
+#!define ADJ_VENTURE_DRILL_DASH_COOLDOWN_BURROWED ADJ_VENTURE_DRILL_DASH_COOLDOWN / 2
+
+rule "[venture/drill_dash.opy]: Constrain Drill Dash Cooldown":
+    @Event eachPlayer
+	@Condition eventPlayer.getCurrentHero() == Hero.VENTURE
+    @Condition not 0 == eventPlayer.getAbilityCooldown(Button.SECONDARY_FIRE)
```
Comment (MaxwellJung) [inline src/heroes/venture/drill_dash.opy]:
```suggestion
    @Condition eventPlayer.getAbilityCooldown(Button.SECONDARY_FIRE) > 0
```

**Review by MaxwellJung — CHANGES_REQUESTED** [review]
works well; just need some styling/formatting changes

**Review by MaxwellJung — COMMENTED** [review]
works well; just need some styling/formatting changes

**Comment by MaxwellJung** [comment]
actually I might close this PR because the cooldown slider seems to work already without this workshop code

**Comment by vwade** [comment]
If you load 7HSKW, the cooldown is set to 75% but during a live game it reverts back to 8/4 seconds for some reason. Because of the uncertainty, at least until the reason can be fully known, I wanted to patch in the specific conditions to course-correct during gameplay.

Also, the condition was set to 0 < CD earlier, but that change wouldn't trigger the event. However the true/not-true setting worked as intended. It is a bit hacky, but functional! :)

**Comment by MaxwellJung** [comment]
@ieToastie disable and reenable drill dash in venture hero settings in `7HSKW` and the cooldown gets fixed

**Comment by MaxwellJung** [comment]
closing as the issue can be fixed without workshop code

**Pull request closed without merging**